homeassistant/components/sonos/media_player.py

```diff
@@ -855,6 +866,20 @@ def update_content(self, event=None):
             self._set_favorites()
             self.schedule_update_ha_state()
 
+    def update_battery_info(self, event=None):
+        """Update battery information from Sonos Move."""
+        try:
+            self._battery_info = self.soco.get_battery_info()
+            self.schedule_update_ha_state()
+        except NotSupportedException:
```

Comment:
Can't we check in `async_seen()` whether the device provides battery info instead of handling this as an exception?

Reply:
I think HomeAssistant complained about network on the main thread when I tried getting the battery info inside async_seen, and I did not find a good hint in Pysonos's api to determine whether it supports battery info. There may be a particular bit in the `supported_features: 457279` that could be used, if it's available at the time of `async_seen()`.